File: mobile/lib/models/matiere.dart
// Modèle représentant une matière
class Matiere {
  final int id;
  final String nom;

  Matiere({
    required this.id,
    required this.nom,
  });

  // Crée un objet Matiere depuis le JSON reçu de l'API
  factory Matiere.fromJson(Map<String, dynamic> json) {
    return Matiere(
      id: json['id'],
      nom: json['nom'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
    };
  }
}

File: backend-api/app/Models/Matiere.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

// Modèle représentant une matière (ex: Maths, Français...)
class Matiere extends Model
{
    // Colonnes autorisées à être remplies
    protected $fillable = ['nom'];

    // Une matière peut être enseignée par plusieurs professeurs
    public function professeurs()
    {
        return $this->belongsToMany(Professeur::class, 'professeur_matiere');
    }

    // Une matière a plusieurs notes d'étudiants
    public function notes()
    {
        return $this->hasMany(Note::class);
    }
}